app/services/course/assessment/submission/auto_grading_service.rb

```diff
@@ -38,6 +50,13 @@ def grade_answers(submission)
   def grade_answer(answer)
     raise ArgumentError if answer.changed?
     answer.auto_grade!
+    # Catch errors if answer is in attempting state, caused by a race condition where
+    # a new attempting answer is created while the submission is finalised, but before the
+    # autograding job is executed.
+  rescue IllegalStateError
```

Comment:
> Is there anywhere that needs `auto_grade!` to fail on an `attempting` answer?

Imo, in every other case except this one because of a race condition, because that aligns with the business logic. Currently `auto_grade!` is only being used in 1 other place, which is when a single answer is submitted to be evaluated. In that case, I would prefer to have the `IllegalStateError` as there are cases where we need to limit the number of attempts. 

Other future use cases would come in once we have discussed multiple submissions and reattempting, so in cases where an exception needs to be made, I'd rather have the dev doing an explicit `rescue`